File: TaskScheduler/TaskScheduler/TaskService.cs
namespace TaskScheduler

{
    //This class contains the information for adding/creating, deleting, updating name, updating description, and
    //getting a task. This is the main class that the TaskServiceTest class accesses.
    public class TaskService
    {
        //This is a readonly dictionary that holds the different tasks that are created. This is set to new
        //for creating a new task when adding into the dictionary.
        private readonly Dictionary<string, Task> _tasks = new();

        //This adds a task to the dictionary. This ensures that a new task with the ID that is input into it as
        //a new Dictionary entry. This simply returns the task using the TryAdd to add a new entry into the dictionary.
        public bool AddTask(Task task)
        {
            return _tasks.TryAdd(task.Id, task);
        }

        //This method is used to delete a task. The way that this is done is by checking that the key does not exist.
        //This key is the ID associated with the specific task the user wants to delete. By checking for the key,
        //it knows whether or not to delete a task. If the task ID does not exist, then it will do nothing, but it will
        //if there is and then access the remove function and return true when this is completed.
        public bool DeleteTask(Task task)
        {
            if (!_tasks.ContainsKey(task.Id))
            {
                return false;
            }
            _tasks.Remove(task.Id);

            return true;

        }

        //This task assigns a name to the task when it is changed. By accessing the task's name element, it is
        //allowing for assigning the new name to the Name element.
        public static void UpdateTaskName(Task task, string newName)
        {
            task.Name = newName;
        }

        //This updates the description for the task. The way that this is done is by accessing the description 
        //that was already set and replacing it with what is stored in the newDesc variable.
        public static void UpdateTaskDesc(Task task, string newDesc)
        {
            task.Description = newDesc;
        }

        //This evaluates for getting a task. This looks at the taskId that is put in and checks whether that exists.
        //If it does exist, then it will return the information stored within the task. Otherwise, it has nothing to
        //display.
        public Task? GetTask(string taskId)
        {
            _tasks.TryGetValue(taskId, out Task? task);
            return task;
        }
    }
}

File: Artifact2/Original/src/taskClassPackage/TaskService.java
package taskClassPackage;

import java.util.HashMap;
import java.util.Map;

public class TaskService {
	
	//This creates a map that stores the tasks.
	private final Map<String, Task> tasks;
	
	//A constructor that creates a new hash map that is labeled as tasks.
	public TaskService()
	{
		tasks = new HashMap<>();
	}
	
	//Adds a task to the list by putting information from the taskID getter and the 
	//task itself.
	public boolean AddTask(Task task)
	{
		if(tasks.containsKey(task.GetTaskID()))
		{
			throw new IllegalArgumentException("Task ID already exists.");
		}
		
		tasks.put(task.GetTaskID(), task);
		
		return true;
	}
	
	//Removes a task from the tasks map that is connected to the task_ID.
	public boolean DeleteTask(String task_ID)
	{
		if(tasks.containsKey(task_ID))
		{
			tasks.remove(task_ID);
			
			return true;
		}
		
		return false;
	}
	
	//Updates the name of the task when it is changed and renames it. This uses the
	//task_ID to ensure that the correct name is changed. This checks that our task is not
	//null and, if it isn't, renames it by setting the name.
	public void UpdateTaskName(String task_ID, String updatedName)
	{
		Task task = tasks.get(task_ID);
		
		if(task == null)
		{
			throw new IllegalArgumentException("Task ID does not exist.");
		}
		
		task.SetName(updatedName);
	}
	
	//Updates the task's description. Using the task ID, it checks if our task is 
	//not null. If it is not, it takes the current description and sets it to the new
	//inputs.
	public void UpdateTaskDesc(String task_ID, String newDesc)
	{
		
		Task task = tasks.get(task_ID);
		
		if(task == null)
		{
			throw new IllegalArgumentException("Task ID does not exist.");
		}
		
		task.SetDesc(newDesc);
	}
	
	//This gets a task from the task_ID.
	public Task GetTask(String task_ID)
	{
		
		return tasks.get(task_ID);
	}
}
